### PR TITLE
client-go: update leader election example

### PR DESCRIPTION
**Pull request by kairen**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the leader election example for help developer know how to implement.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

NONE
```release-note
none
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/examples/leader-election/main.go [hunk 1]
```diff
@@ -37,40 +37,58 @@ import (
...
 	klog.InitFlags(nil)
+
+	var kubeconfig string
+	var leaseLockName string
+	var leaseLockNamespace string
+	var id string
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	// we can use pod name as holder identity
+	flag.StringVar(&id, "id", os.Getenv("POD_NAME"), "the holder identity name")
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/examples/leader-election/main.go]: The README doesn't show how to run the example in a pod. I wouldn't leave the default to be empty.

File: staging/src/k8s.io/client-go/examples/leader-election/main.go [hunk 2]
```diff
@@ -37,40 +37,58 @@ import (
...
-	if len(args) != 3 {
-		log.Fatalf("requires three arguments: ID NAMESPACE CONFIG_MAP_NAME (%d)", len(args))
+
+	if id == "" {
+		klog.Fatal("unable to get id (missing POD_NAME environment variable, or id flag).")
 	}
 
-	// leader election uses the Kubernetes API by writing to a ConfigMap or Endpoints
-	// object. Conflicting writes are detected and each client handles those actions
+	// leader election uses the Kubernetes API by writing to a LeaseLock object.
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/examples/leader-election/main.go]: ...by writing to a lock object, which can be a LeaseLock object (preferred), a ConfigMap, or an Endpoints (deprecated) object.

File: staging/src/k8s.io/client-go/examples/leader-election/main.go [hunk 3]
```diff
@@ -96,6 +113,28 @@ func main() {
 		cancel()
 	}()
 
+	// events from Kubernetes
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/examples/leader-election/main.go]: Just remove this comment? It doesn't seem to add much information.

**Comment by kairen** [comment]
Hi @caesarxuchao, I have updated at your suggestion,  could you review again?

File: staging/src/k8s.io/client-go/examples/leader-election/main.go [hunk 4]
```diff
@@ -37,40 +37,59 @@ import (
...
+	if id == "" {
+		klog.Fatal("unable to get id (missing id flag).")
 	}
 
-	// leader election uses the Kubernetes API by writing to a ConfigMap or Endpoints
-	// object. Conflicting writes are detected and each client handles those actions
+	// leader election uses the Kubernetes API by writing to a
+	// lock object, which can be a LeaseLock object (preferred),
+	// a ConfigMap, or an Endpoints (deprecated) object.
+	// conflicting writes are detected and each client handles those actions
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/examples/leader-election/main.go]: s/conflicting/Conflicting

**Review by caesarxuchao — COMMENTED** [review]
lgtm.

Could you fix the nit and squash the commits? Thanks.

File: staging/src/k8s.io/client-go/examples/README.md [hunk 5]
```diff
@@ -42,6 +42,7 @@ import _ "k8s.io/client-go/plugin/pkg/client/auth/openstack"
   Register a custom resource type with the API, create/update/query this custom
   type, and write a controller that drives the cluster state based on the changes to
   the custom resources.
+- [**Leader election**](./leader-election): Implementing leader election in Kubernetes.
```
[the remark this answers is not in the document]
Comment (kairen) [reply staging/src/k8s.io/client-go/examples/README.md]: Sounds good to me!!

File: staging/src/k8s.io/client-go/examples/leader-election/main.go (hunk 4 above)
Comment (kairen) [reply staging/src/k8s.io/client-go/examples/leader-election/main.go]: ok

**Comment by kairen** [comment]
@caesarxuchao I have fixed the problem for `pull-kubernetes-verify` job, could you check again?

**Review by caesarxuchao — COMMENTED** [review]
/lgtm

**Pull request merged**